src/main/scala/vectorpipe/vectortile/package.scala

```diff
@@ -0,0 +1,89 @@
+  }
+
+  // case class IdFeature[+G <: Geometry, +D](geom: Geometry, data: D, id: Int) extends Feature[G, D](geom, data) {
+  //   override def mapGeom[T <: Geometry](f: G => T): IdFeature[T, D] =
+  //     IdFeature(f(geom), data, id)
+
+  //   override def mapData[T](f: D => T): IdFeature[G, T] =
+  //     IdFeature(geom, f(data), id)
+  // }
+
+  def timedIntersect(geom: Geometry, ex: Extent, id: String)(implicit ec: ExecutionContext) = {
+    val future = Future { geom.intersection(ex) }
+    Try(Await.result(future, 5000 milliseconds)) match {
+      case Success(res) => res
+      case Failure(_) =>
+        logger.warn(s"Could not intersect $geom with $ex [feature id=$id] in 5000 milliseconds")
+        NoResult
+    }
+  }
+
+  case class VTContents(points: List[VectorTileFeature[Point]] = Nil,
+                        multipoints: List[VectorTileFeature[MultiPoint]] = Nil,
+                        lines: List[VectorTileFeature[Line]] = Nil,
+                        multilines: List[VectorTileFeature[MultiLine]] = Nil,
+                        polygons: List[VectorTileFeature[Polygon]] = Nil,
+                        multipolygons: List[VectorTileFeature[MultiPolygon]] = Nil) {
+    def +(other: VTContents) = VTContents(points ++ other.points,
+                                          multipoints ++ other.multipoints,
+                                          lines ++ other.lines,
+                                          multilines ++ other.multilines,
+                                          polygons ++ other.polygons,
+                                          multipolygons ++ other.multipolygons)
+    def +[G <: Geometry](other: VectorTileFeature[G]) = other.geom match {
+      case p : Point        => copy(points=other.asInstanceOf[VectorTileFeature[Point]] :: points)
+      case mp: MultiPoint   => copy(multipoints=mp.asInstanceOf[VectorTileFeature[MultiPoint]] :: multipoints)
+      case l : Line         => copy(lines=l.asInstanceOf[VectorTileFeature[Line]] :: lines)
+      case ml: MultiLine    => copy(multilines=ml.asInstanceOf[VectorTileFeature[MultiLine]] :: multilines)
+      case p : Polygon      => copy(polygons=p.asInstanceOf[VectorTileFeature[Polygon]] :: polygons)
+      case mp: MultiPolygon => copy(multipolygons=mp.asInstanceOf[VectorTileFeature[MultiPolygon]] :: multipolygons)
+    }
+  }
+  object VTContents {
+    def empty() = VTContents()
+  }
+
+  def buildVectorTile[G <: Geometry](geoms: Iterable[VectorTileFeature[G]], ex: Extent, layerName: String, tileWidth: Int): VectorTile = {
+    val contents = geoms.foldLeft(VTContents.empty){ (accum, feature) => accum + feature }
+    val VTContents(pts, mpts, ls, mls, ps, mps) = contents
+    val layer = StrictLayer(
+      name=layerName,
+      tileWidth=tileWidth,
+      version=2,
+      tileExtent=ex,
+      points=pts,
+      multiPoints=mpts,
+      lines=ls,
+      multiLines=mls,
+      polygons=ps, //.sortWith(_.area > _.area),
```

Comment:
Are these sorts something that custom pipelines should be able to do?

Reply:
That's an interesting question.  Is there general utility in this?  I assume this is done to make display more efficient for large tiles, but with the speed of something like MapboxGL, say, does this really come up all that often?  We could default to this behavior, but I have no idea how `area` responds to invalid geometries.  Wouldn't want this op to hang up if there's bad input (which we don't disallow).

Reply:
It might be taken advantage of as a performance improvement, but the reason I was thinking of this is that rendering order is determined by the order of features within a tile. In some cases, it makes sense to render larger features first in order for smaller features to be visible on top of them (especially when other attributes are present and they're rendered differently).

Reply:
`select` can re-order, so I think that covers this.

Reply:
Funny how these kinds of comments come to my attention after I write a possible solution.  We can ask for users to constantly rewrite the sorting logic (if they even think to do so), which I think is something I'd have to look up most every time, or we can provide a flag.  Thoughts?

Reply:
See here: https://github.com/geotrellis/vectorpipe/pull/67/files#diff-c4f36c0e56e8cf0bd8f2338fcac0a47eR39

Reply:
I think I'd prefer a flag, if for no other reason than doing it in `select` would probably generate additional shuffle (vs. doing it on each partition when building the tiles).